File: client/src/game/conf/OddsConfig.ts
class OddsConfig {
    private static _dic:Dictionary<OddsConfig> = new Dictionary<OddsConfig>();
    private static _arr:Array<OddsConfig> = new Array<OddsConfig>();

	 public Id :number ;
    
     public ConfirmValue :number ;
     public MovingRate :number ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<OddsConfig>{
        return  OddsConfig._dic;
    }
    public static get Arr():Array<OddsConfig>{
        return OddsConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:OddsConfig = <OddsConfig>json[i];
				OddsConfig._arr.push(config);
				OddsConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("OddsConfig loader fail!");
        }
    }
}

File: server/src/game/conf/OddsConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type OddsConfigElem struct {
	Id int
	Des string
	ConfirmValue float32
	MovingRate float32
	
}

type OddsConfig0 struct {
	Dic map[int]*OddsConfigElem
	Arr []*OddsConfigElem
}

var (
	OddsConfig OddsConfig0
)

func loadOddsConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	OddsConfig.Dic = make(map[int]*OddsConfigElem)
	OddsConfig.Arr = make([]*OddsConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &OddsConfig.Arr)
	for _,conf := range OddsConfig.Arr{
		OddsConfig.Dic[conf.Id] = conf
	}
	
}
